pkg/arvo/app/publish.hoon

```diff
@@ -484,8 +506,41 @@
   ++  on-watch  on-watch:def
   ++  on-leave  on-leave:def
   ++  on-peek  on-peek:def
-  ++  on-agent  on-agent:def
-  ++  on-arvo  on-arvo:def
+  ++  on-agent  
+    |=  [=wire =sign:agent:gall]
+    ^-  (quip card _this)
+    ?.  ?=([%graph-migrate *] wire)
+      (on-agent:def wire sign)
+    =/  rid=resource
+      (de-path:resource t.wire)
+    ?.  ?=(%watch-ack -.sign)
+      (on-agent:def wire sign)
+    ?~  p.sign
+      ::  if watch acked successfully, then host has completed OTA, and
+      ::  we are safe to add it to the pull-hook
+      :_  this(migrate (~(del by migrate) rid))
+      ~[(poke-graph-pull:main %add entity.rid rid)]
+    ::  if nacked, then set a exponential backoff and retry
+    =/  nack-count=@ud
+      +((~(gut by migrate) rid 0))
+    :_  this(migrate (~(put by migrate) rid nack-count))
+    ::  (bex 19) is roughly 6 days
+    =/  wakeup=@da
+      (add now.bol (bex (min 19 nack-count)))
```

Comment:
Friendly reminder:

```hoon
> `@ux`~s1
0x1.0000.0000.0000.0000
```

```suggestion
    ::  (bex 19) is roughly 6 days
    =/  wakeup=@da
      (add now.bol (mul ~s1 (bex (min 19 nack-count))))
```

You might want to consider lower-bounding nack-count as well, retrying in 1, 4, 8.. etc seconds is unlikely to give different results.

Reply:
Also do we not want to stop trying after, say, 30 days or something?

Reply:
Here's the exponential backoff function in `:acme`:

https://github.com/urbit/urbit/blob/b866ff0e3d0f66ea753553f98530bc801090cd20/pkg/arvo/app/acme.hoon#L423-L431

I think we should put this in zuse somewhere.